src/pages/index.js

```diff
@@ -8,80 +8,172 @@ import { initialCards, cardData, options , cardSelector } from "../utils/constan
+}); 
+
+let cardSection;
+
+const renderCard = (data) => {
+  const card = createCard(data);
+  cardSection.addItem(card);
+};
 
-function createCard(cardData) {
-  const card = new Card(cardData, cardSelector, handleImageClick)
-  return card.getView()
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([userData, cards]) => {
+    userInfo.setUserInfo(userData);
+    cardSection = new Section(
+      {
+        items: cards,
+        renderer: renderCard,
+      },
+      options.cardList
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+const userInfo = new UserInfo(
+  profileTitle, 
+  profileDescription,
+  profileAvatar
+  );
+
+function createCard(data) {
+  const cardElement = new Card(
+    {
+      data,
+      handleImageClick: (imgData) => {
+        imagePreviewPopup.open(imgData);
+      },
+      handleDeleteClick: () => {
+        confirmationPopup.open(() => {
+          confirmationPopup.renderLoading(true);
+          api.deleteCard(data._id)
+            .then(() => {
+              cardElement.removeCard();
+              confirmationPopup.close();
+            })
+            .catch((err) => console.log(`An error occured: ${err}`))
+            .finally(() => confirmationPopup.renderLoading(false));
+        });
+      },
+      handleLikeClick: () => {
+        if (cardElement.isLiked()) {
+          api
+            .removeLike(cardElement._cardId)
+            .then((res) => {
+              cardElement.setLikes(res.isLiked);
+            })
+            .catch((err) => {
+              console.log(`An error occured: ${err}`);
+            });
+        } else {
+          api
+            .addLike(cardElement._cardId)
+            .then((res) => {
+              cardElement.setLikes(res.isLiked);
+            })
+            .catch((err) => {
+              console.log(`An error occured: ${err}`);
+            });
+        }
+      },
+      userId: userInfo.getId(),
+    },
+    options.cardTemplate
+  );
+  return cardElement.getView();
 }
 
+
 function handleProfileEditSubmit(data) {
-  userInfo.setUserInfo(data.name, data.profession)
-  editProfilePopup.close()
+  api.updateProfile(data.name, data.profession)
+  .then((data) => {
+    userInfo.setUserInfo(data)
+  })
+  .catch((err) => {
+    console.log(`An error occured: ${err}`);
+  });
+  editProfilePopup.close();
+}  
+
+function handleAvatarChange(data) {
+  api.updateAvatar(data)
+  .then((data) => {
+  userInfo.setUserInfo(data)
+  })
+  .catch((err) => {
+    console.log(`An error occured: ${err}`);
+  });
+  editProfilePopup.close();
 }  
 
 function handleAddCardSubmit(data){
-  const cardElement = createCard(data);
-  section.addItem(cardElement);
-  addCardPopup.close();
+  api.uploadCard(data)
+    .then((res) => {
+      const card = createCard(res);
+      cardSection.addItem(card);
+      addCardPopup.close();
+    })
+  .catch((err) => {
+    console.log(`An error occured: ${err}`);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors